src/specific/s_input.c

```diff
@@ -858,6 +856,18 @@ static INPUT_STATE S_Input_GetControllerState(
         state.load                   |= S_Input_GetBindState(INPUT_ROLE_LOAD, cntlr_layout_num);
         state.toggle_fps_counter     |= S_Input_GetBindState(INPUT_ROLE_FPS, cntlr_layout_num);
         state.toggle_bilinear_filter |= S_Input_GetBindState(INPUT_ROLE_BILINEAR, cntlr_layout_num);
+        state.menu_up                |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_UP);
+        state.menu_down              |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_DOWN);
+        state.menu_left              |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_LEFT);
+        state.menu_right             |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_RIGHT);
+        if (m_ControllerType == SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_PRO || m_ControllerType == SDL_CONTROLLER_TYPE_NINTENDO_SWITCH_JOYCON_PAIR) {
+            state.menu_confirm           |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+            state.menu_back              |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+        } else {
+            state.menu_confirm           |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+            state.menu_back              |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+        }
```

Comment:
To be honest I'd forgo this condition in favor of the old code:

```c
        state.menu_confirm            |= S_Input_GetBindState(INPUT_ROLE_ACTION, cntlr_layout_num);
        state.menu_back               |= S_Input_GetBindState(INPUT_ROLE_OPTION, cntlr_layout_num);
```

or even remove all `menu` assignments altogether from the controller code, as it'll be augmented in `Input_GetState`.

Asking for @lahm86 's opinion on this.

Reply:
The old style seems easier to read - I assume it achieves the same result?

Reply:
Ya I can revert this for controller since we don't really need it like we do for keyboard.

Reply:
Oh btw the reason this condition was there was because the switch uses different confirm buttons compared to Xbox / PS. But best to be consistent.